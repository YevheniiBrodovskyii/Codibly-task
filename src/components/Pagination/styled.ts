import styled from 'styled-components'

export const PaginationTitle = styled.p`
	margin: 0;
	text-align: center;
	font-size: 25px;
`

export const PaginationContainer = styled.div`
	display: flex;
	justify-content: space-between;
	width: 86px;
	margin: 0 auto;
	background-color: black;
	border-radius: 10px;
	padding: 10px;
`

export const PaginationBtn = styled.button`
	background-color: unset;
	border: none;
	color: ${({ disabled }) => (disabled ? 'unset' : 'white')};
	cursor: pointer;
	transition: 0.3s all;
	&:hover {
		transform: scale(1.3);
	}
`
