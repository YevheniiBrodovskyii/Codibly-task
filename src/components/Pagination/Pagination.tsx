import { FC } from 'react'
import { useAppSelector } from '../../hook'

import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos'
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos'

import { PaginationTitle, PaginationBtn, PaginationContainer } from './styled'

const Pagination: FC<{
	setPage: (page: string) => void
}> = ({ setPage }) => {
	let pageNumber = useAppSelector((state) => state.products.list.page)
	const totalPages = useAppSelector((state) => state.products.list.total_pages)

	return (
		<>
			<PaginationTitle>Page {pageNumber}</PaginationTitle>
			<PaginationContainer>
				<PaginationBtn
					onClick={() => {
						setPage((pageNumber -= 1).toString())
					}}
					disabled={pageNumber <= 1}
				>
					<ArrowBackIosIcon fontSize="large" style={{ float: 'left' }} />
				</PaginationBtn>

				<PaginationBtn
					onClick={() => {
						setPage((pageNumber += 1).toString())
					}}
					disabled={pageNumber >= totalPages}
					data-testid="pagination-right-arrow"
				>
					<ArrowForwardIosIcon fontSize="large" style={{ float: 'right' }} />
				</PaginationBtn>
			</PaginationContainer>
		</>
	)
}

export default Pagination
