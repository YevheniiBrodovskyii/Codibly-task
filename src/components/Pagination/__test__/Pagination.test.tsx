import Pagination from '../Pagination'

import {
	screen,
	fireEvent,
	waitFor,
	RenderResult,
} from '@testing-library/react'

import { renderWithProviders } from '../../../utils/test-utils'

describe('Pagination tests', () => {
	let component: RenderResult
	const mockfetch = jest.fn()

	beforeEach(() => {
		component = renderWithProviders(<Pagination setPage={mockfetch} />)
	})

	it('should match snapshot on initial render', () => {
		expect(component).toMatchSnapshot()
	})

	it('should trigger fetch arrow button', async () => {
		console.log(await screen.getByTestId('pagination-right-arrow'))
		console.log('Clicking')
		fireEvent.click(screen.getByTestId('ArrowForwardIosIcon'))
		await waitFor(() => expect(mockfetch).toHaveBeenCalledTimes(1))
		expect(mockfetch).toBeCalledWith(1)
	})
})
