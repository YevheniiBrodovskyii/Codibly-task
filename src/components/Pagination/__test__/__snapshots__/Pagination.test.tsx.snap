// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination tests should match snapshot on initial render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bcXHqe crTARc"
      >
        <button
          class="sc-gswNZR eQwewK"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="ArrowBackIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
            />
          </svg>
        </button>
        <button
          class="sc-gswNZR eQwewK"
          data-testid="pagination-right-arrow"
          disabled=""
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
            data-testid="ArrowForwardIosIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bcXHqe crTARc"
    >
      <button
        class="sc-gswNZR eQwewK"
        disabled=""
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="ArrowBackIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
          />
        </svg>
      </button>
      <button
        class="sc-gswNZR eQwewK"
        data-testid="pagination-right-arrow"
        disabled=""
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
